ui/main/src/app/modules/feedconfiguration/feedconfiguration.component.ts

```diff
@@ -116,21 +134,33 @@ export class FeedconfigurationComponent implements OnInit {
                 const statesArray: { stateLabel: string, stateControlIndex: number }[]
                     = new Array<{stateLabel: string, stateControlIndex: number}>();
 
-                const processI18n = (!!process.name) ? this.getI18nPrefixFromProcess(process) + process.name :
+                let processLabel = (!!process.name) ? this.getI18nPrefixFromProcess(process) + process.name :
                     this.getI18nPrefixFromProcess(process) + process.id;
+                this.translateService.get(processLabel).subscribe(translate => { processLabel = translate; });
 
                 for (const key in process.states) {
                     const value = process.states[key];
-                    const stateLabel = (!!value.name) ? this.getI18nPrefixFromProcess(process) + value.name :
+                    let stateLabel = (!!value.name) ? this.getI18nPrefixFromProcess(process) + value.name :
                         this.getI18nPrefixFromProcess(process) + key;
 
+                    this.translateService.get(stateLabel).subscribe(translate => { stateLabel = translate; });
+
                     statesArray.push({stateLabel: stateLabel, stateControlIndex: stateControlIndex});
                     this.preparedListOfProcessesStates.push({
                         processId: process.id,
                         stateId: key});
                     stateControlIndex++;
                 }
-                this.processesStatesLabels.set(process.id, {processI18n: processI18n,
+                statesArray.sort((obj1, obj2) => {
+                    if (obj1.stateLabel > obj2.stateLabel) {
+                        return 1;
+                    }
+                    if (obj1.stateLabel < obj2.stateLabel) {
+                        return -1;
+                    }
+                    return 0;
```

Comment:
Avoid duplication : Create a method sortByAlphabeticalOrder and send this method as parameter for statesArray.sort

Reply:
Done.